osbenchmark/builder/models/plugin_config_instance.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass, field
+from typing import List
+
+
+@dataclass
+class PluginConfigInstance:
+    # name of the initial Python file to load for plugins.
+    ENTRY_POINT = "plugin"
+
+    """
+    Creates new settings for a plugin attached to a benchmark candidate.
+
+    :param names: Descriptive name for this plugin_config_instance.
+    :param is_core_plugin: A boolean dictating if the plugin is a core plugin.
+    :param config_names: A list of config folder names where the raw config can be found. May be ``None``.
+    :param root_path: The root path from which bootstrap hooks should be loaded if any. May be ``None``.
+    :param config_paths: A non-empty list of paths where the raw config can be found.
+    :param variables: A dict containing variable definitions that need to be replaced.
+    """
+    name: str
+    is_core_plugin: bool = False
+    config_names: List[str] = None
+    root_path: str = None
+    config_paths: List[str] = field(default_factory=list)
+    variables: dict = field(default_factory=dict)
+
+    @staticmethod
+    def get_entry_point():
+        return PluginConfigInstance.ENTRY_POINT
+
+    def __str__(self):
+        return "Plugin descriptor for [%s]" % self.name
+
+    def __repr__(self):
+        r = []
+        for prop, value in vars(self).items():
+            r.append("%s = [%s]" % (prop, repr(value)))
+        return ", ".join(r)
+
+    def __hash__(self):
+        return hash(self.name) ^ hash(self.config_names) ^ hash(self.is_core_plugin)
```

Comment:
Any consideration for not including ``config_paths``, ``root_path`` and  ``variables`` in the hashing?

Reply:
I think the 3 used are sufficient to uniquely identify a plugin config instance. Not entirely sure tho as it's inherited from a copy-paster